### PR TITLE
fix: do not filter out removable devices

### PR DESCRIPTION
**Pull request by leelavg**
- hot pluggable devices are represented as removable in lsblk o/p
- not all removable devices should be filtered out and this PR removes
  the existing filter
- as a side effect ROM is being picked up for VG ops and that is
  filtered based on Type of device

fixes: #109

Signed-off-by: Leela Venkaiah G <lgangava@redhat.com>

### CODE REVIEW

**Comment by leelavg** [comment]
- filtering out ROM device
``` json
{"level":"info","ts":1649414296.9406986,"logger":"controller.lvmvolumegroup.vg-manager","msg":"does not match filter","reconciler group":"lvm.topolvm.io","reconciler kind":"LVMVolumeGroup","name":"vg1","namespace":"lvm-operator-system","Device.Name":"sr0","filter.Name":"isUsableByDeviceType"}
```
- no regression
``` json
{"level":"info","ts":1649415065.6944656,"logger":"controller.lvmvolumegroup.vg-manager","msg":"does not match filter","reconciler group":"lvm.topolvm.io","reconciler kind":"LVMVolumeGroup","name":"vg1","namespace":"lvm-operator-system","Device.Name":"loop4","filter.Name":"isUsableByDeviceType"}
```

**Review by sp98 — APPROVED** [review]

File: pkg/vgmanager/filter.go [hunk 1]
```diff
@@ -25,13 +25,12 @@ import (
...
 	// filter names:
 	notReadOnly           = "notReadOnly"
-	notRemovable          = "notRemovable"
 	notSuspended          = "notSuspended"
 	noBiosBootInPartLabel = "noBiosBootInPartLabel"
 	noFilesystemSignature = "noFilesystemSignature"
 	noBindMounts          = "noBindMounts"
 	noChildren            = "noChildren"
-	usableLoopDevice      = "usableLoopDevice"
+	isUsableByDeviceType  = "isUsableByDeviceType"
```
Comment (nbalacha) [inline pkg/vgmanager/filter.go:33]:
```suggestion
	usableDeviceType  = "usableDeviceType"
```

**Review by nbalacha — CHANGES_REQUESTED** [review]
If, in future, there is a requirement to be able to configure the VG to not use loop devices , how would this be implemented?

Would it better to have one filter per type and add them to the list of filters for a VG dynamically depending on the configuration?

**Comment by leelavg** [comment]
> how would this be implemented?
- then it'll be a straightforward implementation, we use all the strings based on device types in single `case` of `switch` and will filter them out
- it basically boils out to filtering out devices based on device types currently loop is a special case

> Would it better to have one filter per type
- I feel adding filters per type will be too verbose and only increases filter count

**Comment by nbalacha** [comment]
> > how would this be implemented?
> 
> * then it'll be a straightforward implementation, we use all the strings based on device types in single `case` of `switch` and will filter them out
> * it basically boils out to filtering out devices based on device types currently loop is a special case
> 
Can you elaborate on this seeing that the supported types would be dynamic?
> > Would it better to have one filter per type
> 
> * I feel adding filters per type will be too verbose and only increases filter count

**Comment by nbalacha** [comment]
> > > how would this be implemented?
> > 
> > 
> > 
> > * then it'll be a straightforward implementation, we use all the strings based on device types in single `case` of `switch` and will filter them out
> > * it basically boils out to filtering out devices based on device types currently loop is a special case
> > 
> > > Would it better to have one filter per type
> > 
> > 
> > 
> > * I feel adding filters per type will be too verbose and only increases filter count
> 
> Agree with Leela here. I don't think we need multiple filters for each device class. The current switch case usage would suffice.

If, in future, we want to allow dynamic device type config, the switch case will not work. Say we want to use loopDevices for one cluster and not the other, we need a way to dynamically determine the filters.

**Review by nbalacha — APPROVED** [review]
Approving for now but we should relook at the filter approach.

**Pull request merged**